### PR TITLE
Pokemon SS Isle of Armor DLC

### PR DESCRIPTION
**Pull request by AustinXII**

### CODE REVIEW

File: calc/src/data/moves.ts [hunk 1]
```diff
@@ -2891,7 +2891,24 @@ const SS_PATCH: {[name: string]: DeepPartial<MoveData>} = {
...
+    bp: 70,
+    type: 'Dragon',
+    target: 'allAdjacentFoes',
+    category: 'Special',
+    secondaries: true,
+  },
   'Clangorous Soul': {bp: 0, type: 'Dragon', isSound: true},
+  'Coaching': {
+    bp: 0,
+    type: 'Normal',
```
Comment (KrisXV) [inline calc/src/data/moves.ts:2904]:
```suggestion
    type: 'Fighting',
```

File: calc/src/data/moves.ts [hunk 2]
```diff
@@ -2891,7 +2891,24 @@ const SS_PATCH: {[name: string]: DeepPartial<MoveData>} = {
...
+    category: 'Status',
+  },
+  'Corrosive Gas': {
+    bp: 0,
+    type: 'Normal',
```
Comment (KrisXV) [inline calc/src/data/moves.ts:2909]:
```suggestion
    type: 'Poison',
```

File: calc/src/data/moves.ts [hunk 3]
```diff
@@ -3337,6 +3486,36 @@ const SS_PATCH: {[name: string]: DeepPartial<MoveData>} = {
...
     maxPower: 130,
   },
+  'Surging Strikes': {
+    bp: 25,
+    type: 'Water',
+    category: 'Physical',
+    makesContact: true,
+    multihit: 3,
+  },
+  'Terrain Pulse': {
```
Comment (KrisXV) [inline calc/src/data/moves.ts:3496]: add `isPulse` to this

File: calc/src/data/moves.ts [hunk 4]
```diff
@@ -3337,6 +3486,36 @@ const SS_PATCH: {[name: string]: DeepPartial<MoveData>} = {
...
+    bp: 50,
+    type: 'Normal',
+    category: 'Special',
+    makesContact: true,
```
Comment (KrisXV) [inline calc/src/data/moves.ts:3500]: no it doesnt

File: calc/src/data/moves.ts [hunk 5]
```diff
@@ -3337,6 +3486,36 @@ const SS_PATCH: {[name: string]: DeepPartial<MoveData>} = {
     zp: 175,
     maxPower: 130,
   },
+  'Surging Strikes': {
```
Comment (KrisXV) [inline calc/src/data/moves.ts:3489]: add `willCrit` to this

File: calc/src/data/species.ts [hunk 6]
```diff
@@ -8014,6 +8041,12 @@ const SS_PATCH: {[name: string]: DeepPartial<SpeciesData>} = {
     abilities: {0: 'Hyper Cutter'},
     baseSpecies: 'Kingler',
   },
+  'Kubfu': {
+    types: ["Fighting"],
+    bs: {hp: 60, at: 90, df: 60, sa: 53, sd: 50, sp: 73},
```
Comment (KrisXV) [inline calc/src/data/species.ts:8046]:
```suggestion
    bs: {hp: 60, at: 90, df: 60, sa: 53, sd: 50, sp: 72},
```

File: calc/src/data/species.ts [hunk 7]
```diff
@@ -8142,6 +8175,13 @@ const SS_PATCH: {[name: string]: DeepPartial<SpeciesData>} = {
     abilities: {0: 'Static'},
     baseSpecies: 'Pikachu',
   },
+  'Pikachu-World': {
```
Comment (KrisXV) [inline calc/src/data/species.ts:8178]: add this to Pikachu's `otherFormes`

File: calc/src/data/species.ts [hunk 8]
```diff
@@ -8191,6 +8231,14 @@ const SS_PATCH: {[name: string]: DeepPartial<SpeciesData>} = {
     bs: {hp: 100, at: 125, df: 90, sa: 60, sd: 70, sp: 85},
     weightkg: 90,
     abilities: {0: 'Overgrow'},
+    otherFormes: ['Rillaboom-Gmax'],
+  },
+  'Rillaboom-Galar': {
```
Comment (KrisXV) [inline calc/src/data/species.ts:8236]:
```suggestion
  'Rillaboom-Gmax': {
```

lol

File: calc/src/data/species.ts [hunk 9]
```diff
@@ -8278,6 +8326,13 @@ const SS_PATCH: {[name: string]: DeepPartial<SpeciesData>} = {
     abilities: {0: 'Cheek Pouch'},
     nfe: true,
   },
+  'Slowbro-Galar': {
+    types: ['Psychic', 'Poison'],
+    bs: {hp: 95, at: 100, df:95, sa: 100, sd: 70, sp: 30},
```
Comment (KrisXV) [inline calc/src/data/species.ts:8331]:
```suggestion
    bs: {hp: 95, at: 100, df: 95, sa: 100, sd: 70, sp: 30},
```

File: calc/src/data/species.ts [hunk 10]
```diff
@@ -7479,13 +7480,16 @@ const SS_PATCH: {[name: string]: DeepPartial<SpeciesData>} = {
       'Pikachu-Partner',
       'Pikachu-Sinnoh',
       'Pikachu-Unova',
+      'Pikachu-World',
     ],
   },
   Ponyta: {otherFormes: ['Ponyta-Galar']},
   Rapidash: {otherFormes: ['Rapidash-Galar']},
+  Slowbro: {otherFormes: ['Slowbro-Galar']},
```
Comment (KrisXV) [inline calc/src/data/species.ts:7488]:
```suggestion
  Slowbro: {otherFormes: ['Slowbro-Galar', 'Slowbro-Mega']},
```

File: calc/src/mechanics/gen78.ts [hunk 11]
```diff
@@ -87,7 +87,10 @@ export function calculateSMSS(
     return result;
   }
 
-  if (field.defenderSide.isProtected && !move.breaksProtect && !move.isZ && !attacker.isDynamaxed) {
+  const breaksProtect = move.breaksProtect || move.isZ || attacker.isDynamaxed
+  || attacker.hasAbility('Unseen Fist');
```
Comment (KrisXV) [inline calc/src/mechanics/gen78.ts:91]:
```suggestion
  || (attacker.hasAbility('Unseen Fist') && move.makesContact);
```

File: calc/src/mechanics/gen78.ts [hunk 12]
```diff
@@ -361,10 +371,27 @@ export function calculateSMSS(
     basePower = r >= 4 ? 150 : r >= 3 ? 120 : r >= 2 ? 80 : r >= 1 ? 60 : 40;
     desc.moveBP = basePower;
     break;
+  case 'Expanding Force':
+    basePower = move.bp * (field.hasTerrain("Psychic") ? 2 : 1);
+    move.target = field.hasTerrain("Psychic") ? "allAdjacentFoes" : "normal";
+    desc.moveBP = basePower;
+    break;
```
Comment (KrisXV) [inline calc/src/mechanics/gen78.ts:378]: This only applies when the user is grounded (aka not holding air balloon/doesnt have levitate/not flying type), idk if there's a way to check that

this is the only move it's confirmed to apply to for now

File: calc/src/mechanics/gen78.ts [hunk 13]
```diff
@@ -414,6 +445,10 @@ export function calculateSMSS(
     basePower = field.weather && !field.hasWeather('Strong Winds') ? 100 : 50;
     desc.moveBP = basePower;
     break;
+  case 'Weather Ball':
```
Comment (KrisXV) [inline calc/src/mechanics/gen78.ts:448]:
```suggestion
  case 'Terrain Pulse':
```

File: calc/src/mechanics/gen78.ts (hunk 12 above)
Comment (AustinXII) [reply calc/src/mechanics/gen78.ts:378]: we have `isGrounded`

File: calc/src/mechanics/gen78.ts (hunk 11 above)
Comment (AustinXII) [reply calc/src/mechanics/gen78.ts:91]: `move.flags.contact`

File: calc/src/data/species.ts (hunk 7 above)
Comment (AustinXII) [reply calc/src/data/species.ts:8178]: it is

**Pull request merged**